src/Validator/Rules/ValuesOfCorrectType.php

```diff
@@ -33,8 +35,15 @@
  */
 class ValuesOfCorrectType extends AbstractValidationRule
 {
-    static function badValueMessage($typeName, $valueName, $message = null)
+    private static $fieldName;
```

Comment:
We shouldn't rely on static $fieldName here. In theory, you may have multiple rules running in parallel which would break with static variable here. See below for suggestions.

Reply:
Resolved in 9dc41f0